### PR TITLE
chore: Update to twilight 0.4 and metrics 0.16

### PR DESCRIPTION
**Pull request by Gelbpunkt**
This allows for using webpki-roots to build a fully static and independent binary in Docker and not rely on Alpine for the final image, further reducing the size.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -76,13 +74,21 @@ RUN source $HOME/.cargo/env && \
     cp target/$RUST_TARGET/release/twilight-http-proxy /twilight-http-proxy && \
     actual-strip /twilight-http-proxy
 
-FROM docker.io/${FINAL_TARGET}/alpine:latest
-ARG TARGET
+FROM docker.io/library/alpine:edge AS dumb-init
```
Comment (7596ff) [inline Dockerfile:77]: We should pick a different name for this concept than "dumb"

**Review by 7596ff — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (Gelbpunkt) [reply Dockerfile:77]: The program is actually called like that and is a fully static init system. See https://github.com/Yelp/dumb-init

Comment (7596ff) [reply Dockerfile:77]: Oh. lol

**Review by 7596ff — COMMENTED** [review]
Clearing my review

**Comment by AEnterprise** [comment]
didn't 0.4 add some new routes that the proxy needs names for? like search members endpoint and stage endpoints

**Comment by Gelbpunkt** [comment]
> didn't 0.4 add some new routes that the proxy needs names for? like search members endpoint and stage endpoints

oops yeah, they did not show when I tested because of the non_exhaustive, fixing this

**Comment by Gelbpunkt** [comment]
done, all new routes have been added

**Review by AEnterprise — APPROVED** [review]

**Review by spring4175 — APPROVED** [review]

**Pull request merged**